util/src/test/java/com/google/cloud/hadoop/util/RetryHttpInitializerTest.java

```diff
@@ -166,4 +246,23 @@ private static RetryHttpInitializer createRetryHttpInitializer(Credentials crede
             .setReadTimeout(Duration.ofSeconds(5))
             .build());
   }
+
+  // Helper class which help provide a custom test implementation of RequestTracker
```

Comment:
I would recommend adding a note on limitation of `TestRetryHttpInitializer`, which is, "it can only used against test with only one request because RequestTracker is initialized along with HttpInitializer".

Reply:
This is an inner class and is an implementation detail. I think it is obvious from the code that this applies for one request per TestRetryHttpInitializer.